cpp/sqliteBridge.cpp

```diff
@@ -130,29 +130,44 @@ SQLiteOPResult sqliteCloseDb(string const dbName)
   };
 }
 
-// SequelResult sequel_attach(string const &dbName)
-//{
-//     if(dbMap.count(dbName) == 0){
-//         cout << "[react-native-quick-sqlite]: DB " << dbName << " not open" << endl;
-//         return SequelResult{
-//             SequelResultError,
-//             dbName + " is not open",
-//             jsi::Value::undefined()
-//         };
-//     }
-
-// TODO: What does "Attach" do? is it really necessary?
-// https://github.com/andpor/react-native-sqlite-storage/blob/master/platforms/ios/SQLite.m#L362
-
-//    NSString* sql = [NSString stringWithFormat:@"ATTACH DATABASE '%@' AS %@", dbPathToAttach, dbAlias];
-//
-//            if(sqlite3_exec(db, [sql UTF8String], NULL, NULL, NULL) == SQLITE_OK) {
-//              pluginResult = [SQLiteResult resultWithStatus:SQLiteStatus_OK messageAsString:@"Database attached successfully."];
-//            } else {
-//              pluginResult = [SQLiteResult resultWithStatus:SQLiteStatus_ERROR messageAsString:@"Unable to attach DB"];
-//            }
-
-//}
+SQLiteOPResult sqliteAttachDb(string const mainDBName, string const docPath, string const databaseToAttach, string const alias)
+{
+  /**
+   * There is no need to check if mainDBName is opened because sqliteExecuteLiteral will do that.
```

Comment:
Is this really correct? the executeSQL functions check for the in memory map of opened databases at the top of this file. Doesn't attaching mean the database needs to be placed in that map as well?

Reply:
> Is this really correct? the executeSQL functions check for the in memory map of opened databases at the top of this file. Doesn't attaching mean the database needs to be placed in that map as well?

I believe that this is the correct approach, since attached databases does not have their own connection. They are aggregated to the main database connection and must be accessed through it with their aliases.

Reply:
👍